File: case_studies/tuto/skewing/elim_loop_single/elim_loop_single.cpp

int main() {
  for (int k = 0; k < 0 + 1; k++) {
    const int j = 12 + k;
  }
}
// Expected to fail : No for loop with +1 increment
int main2() {

  for (int k = 10; k < 25; k++) {
    int j = 12;
  }
  return 12;
}

File: tests/loop/elim_loop_single/elim_loop_single.ml
open Optitrust
open Prelude
let _ =
  Run.script_cpp (fun _ ->
    !! Loop.elim_loop_single [cFunBody "main"; cFor "i"];
    !! Loop.elim_loop_single [cFunBody "main2"; cFor "i"];
    !! Trace.failure_expected
        (fun _e -> true)
        (fun _ -> Loop.elim_loop_single [cFunBody "main3"; cFor "i"];)
  )
